llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -2532,7 +2532,8 @@ bool CodeGenPrepare::dupRetToEnableTailCallOpts(BasicBlock *BB,
     }
 
     PN = dyn_cast<PHINode>(V);
-    if (!PN)
+
+    if (isa<Constant>(V))
```

Comment:
Could you please move this line before the `dyn_cast<PHINode>` call?

Reply:
May be worth it to anticipate before the `dyn_cast<BitCastInst>`?